source/MRMesh/MRObjectMesh.cpp

```diff
@@ -249,14 +294,40 @@ std::shared_ptr<ObjectMesh> merge( const std::vector<std::shared_ptr<ObjectMesh>
                     faceColors[mergeId] = curColorMap.size() <= thisId ? obj->getFrontColor() : curColorMap[thisId];
             }
         }
+        if ( needSaveTexture )
+        {
+            const auto& curUvCoords = obj->getUVCoords();
+            for ( VertId thisId = 0_v; thisId < vertMap.size(); ++thisId )
+            {
+                if ( auto mergeId = vertMap[thisId] )
+                    uvCoords[mergeId] = curUvCoords[thisId];
+            }
+            const auto curTextures = obj->getTextures();
+            textures.vec_.insert( textures.vec_.end(), curTextures.vec_.begin(), curTextures.vec_.end() );
+            TextureId previousNumTexture;
+            if ( numObject == 0 )
+                previousNumTexture = TextureId( 0 );
+            else
+                previousNumTexture = TextureId( numTextures[ numObject - 1 ] );
```

Comment:
mb just update curent texture id on each iteration? I think`numTextures` is not needed